AEX-2 Compatibility

petarbaykov opened the pull request:
- Connection between aepp and wallet
- init RpcWallet
- listen for changes in address and network
- open connection popup

Aex-2 communication can be tested with this aepp -> http://aex2.aepps.tech/

There are several things that I saw while integrating sdk-s `RpcWallet` and `RpcAepp` that are connected to the sdk implementation of aex-2 (https://github.com/aeternity/aepp-sdk-js/pull/512)
- communication between the aepp and extension is done via `chrome.runtime.connect` method inside the aepp. This method will connect to the extension only if the domain of the aepp is specified in manifest.json of the extension (https://developer.chrome.com/apps/manifest/externally_connectable). I think content script should be in the middle of the aepp and the extension. This way it will work no matter what is the domain of the aepp
- The object initiated in aepp is called `RpcAepp`. I think the object should be called `Aepp` and get the properties of the `RpcAepp` if extension wallet is available. Because now when I run the example aepp app it doesnt work with base aepp because the aepp initialize `RpcAepp` object and not an `Aepp` object. Another thing is that when you initiate `RpcAepp` object you need to specify network, I think the network should be retrieved from the wallet extension
- Deploying smart contracts throws an error at this point with the current fork of the sdk used in this pull request. I think this is because of the aex-2 branch of the sdk not being up-to date with `develop`.

petarbaykov committed:
Working on AEX-2 compitability

petarbaykov committed:
Remove unnecessary files

petarbaykov committed:
Aex-2 compitability

petarbaykov committed:
RpcWallet

- Connection between aepp and wallet
- init RpcWallet
- listen for changes in address and network
- open connection popup

mradkov commented on package.json:
```diff
@@ -22,20 +22,21 @@
     "@aeternity/aepp-components": "git+https://github.com/aeternity/aepp-components.git#develop",
     "@aeternity/aepp-components-3": "3.0.0-alpha.9",
-    "@aeternity/aepp-sdk": "6.0.0",
+    "@aeternity/aepp-sdk": "github:aeternity/aepp-sdk-js#aex-2",
```
This branch is not up-to date with the Lima release and results in `Unable to connect to choosen node` error.

mradkov replied:
Are the new changes working with the branch of the pull request to the sdk that supports aex-2 (https://github.com/aeternity/aepp-sdk-js/pull/512/)?

petarbaykov replied:
The changes are working with fork of the sdk aex-2 branch. The only difference is changed node version because i was not able to connect to the testnet -> https://github.com/petbaik/aepp-sdk-js/blob/feature/aex-2/es/node.js#L158
` "@aeternity/aepp-sdk": "github:petbaik/aepp-sdk-js#feature/aex-2"`
The only difference between  sdk aex-2 branch and my fork is the changed node version

petarbaykov committed:
Aex2

- Open connection popup and confirm or reject connection with aepp
- Open tx sign popup

petarbaykov committed:
Sign tx popup

petarbaykov committed:
Aex-2 comunication

petarbaykov committed:
Update inject.js

petarbaykov committed:
Update initRpcWallet.js

petarbaykov committed:
Replace chrome with browser in background script

mradkov commented:
After a discussion with the SDK team, we agreed that the [AEX-2](https://github.com/aeternity/AEXs/blob/master/AEXS/aex-2.md) functionality will be merged in the master branch after it is being `finalized`.

So we put this on hold until then.

mradkov commented:
https://github.com/aeternity/aepp-sdk-js/pull/512

Is going to be included in v7.0.0 major release of the sdk-js.

mradkov commented:
> [aeternity/aepp-sdk-js#512](https://github.com/aeternity/aepp-sdk-js/pull/512)
> 
> Is going to be included in v7.0.0 major release of the sdk-js.

Merged. Ref. https://github.com/aeternity/aepp-sdk-js/commit/e7e9eb49dc58a8dad0e6936318dd40b7d3a39f25

mradkov commented:
Ref. implementation here: https://github.com/aeternity/aepp-sdk-js/tree/develop/examples/browser/extension

petarbaykov committed:
Update SDK version to develop branch, compatibility with changes in SDK's AEX-2 implementation

petarbaykov commented:
> > [aeternity/aepp-sdk-js#512](https://github.com/aeternity/aepp-sdk-js/pull/512)
> > Is going to be included in v7.0.0 major release of the sdk-js.
> 
> Merged. Ref. [aeternity/aepp-sdk-js@e7e9eb4](https://github.com/aeternity/aepp-sdk-js/commit/e7e9eb49dc58a8dad0e6936318dd40b7d3a39f25)

The object initialized in aepp is still called RpcAepp. The aepps created so far still will not be compatible with the extension unless they use RpcAepp.

mradkov commented:
> communication between the aepp and extension is done via chrome.runtime.connect method inside the aepp. This method will connect to the extension only if the domain of the aepp is specified in manifest.json of the extension (https://developer.chrome.com/apps/manifest/externally_connectable). I think content script should be in the middle of the aepp and the extension. This way it will work no matter what is the domain of the aepp

After investigating this I came to the conclusion we cannot use wildcard matches for the externally connectable, so what you've proposed with the content script in the middle is a good approach and maybe the only feasible one at this point in time. 

Together with @nduchak we discussed that and we think we can remove the `Aepp` and stick to using post messages forwarding using the interface proposed in AEX2. In order to be compatible an application will need to import the `RpcAepp` rather than `Aepp` which will be the recommended way forward.

mradkov commented:
https://github.com/aeternity/aepp-sdk-js/pull/815
